example-client/client.py

```diff
@@ -64,43 +64,76 @@ def run_structure(input: str) -> Optional[str]:
         args=[input],
     )
 
-    # Runs are asynchronous, so we need to poll the status until it's no longer running.
     structure_run_id = structure_run["structure_run_id"]
-    status = structure_run["status"]
-    printed_event_ids = set()  # Keep track of which events we've printed.
-    while status not in ("SUCCEEDED", "FAILED"):
-        structure_run = get_structure_run(
-            host=HOST, api_key=GT_API_KEY, run_id=structure_run_id
-        )
-        status = structure_run["status"]
 
-        # You can comment out this block if you don't want to stream events as they occur.
+    received_event_ids = set()  # Keep track of which events we've already received.
+    structure_has_terminal_event = False
+
+    # The client will continuously poll the Structure Run until it receives one of three events:
+    # 1. Our Structure has voluntarily emitted a FinishStructureRunEvent
+    # 2. The Structure has stopped running due to an error
+    # 3. The Structure has stopped running without an error
+    while not structure_has_terminal_event:
+        # TODO: Can the API be adjusted to give us all NEW events that occurred after a specific point?
         event_list = get_structure_run_events(
             host=HOST, api_key=GT_API_KEY, run_id=structure_run_id
         )
         events = event_list["events"]
-        printed_event_ids = print_streaming_events(events, printed_event_ids)
 
-        time.sleep(1)  # Poll every second.
+        # Find all events that are new to us.
+        for event in events:
+            event_id = event["event_id"]
+            if event_id not in received_event_ids:
+                # Mark it as received.
+                received_event_ids.add(event_id)
+
+                # Is this an event type that we care about?
+                event_type = event["value"]["type"]
+
+                match event_type:
+                    case "CompletionChunkEvent":
+                        # If the agent is set to streaming, print out each streaming chunk
+                        completion_token = event["value"]["token"]
+                        print(completion_token, flush=True, end="")
+                    case "FinishStructureRunEvent":
+                        # Our Griptape Structure told us that it was done, and gave us a payload for the output. Our client can stop polling the Structure.
+                        structure_has_terminal_event = True
+                        structure_output = event["value"]["output_task_output"]
+                        print(structure_output)
```

Comment:
I think these should be tucked under some sort of `FrameworkEvent.`

Reply:
where does that tucking happen? In the Framework, or in the service as events are ejected?

Reply:
It'd happen in `POST /events` before they're stored in the DB.